src/test/java/com/fasterxml/jackson/failing/JsonTypeInfoCaseInsensitive1983Test.java

```diff
@@ -0,0 +1,44 @@
+package com.fasterxml.jackson.failing;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+
+public class JsonTypeInfoCaseInsensitive1983Test extends BaseMapTest {
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "operation")
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = Equal.class, name = "eq"),
+            @JsonSubTypes.Type(value = NotEqual.class, name = "noteq"),
+    })
+    static abstract class Filter {
+    }
+
+    static class Equal extends Filter {
+    }
+
+    static class NotEqual extends Filter {
+    }
+
+    public void testReadMixedCaseSubclass() throws IOException {
+//        There is also "ACCEPT_CASE_INSENSITIVE_VALUES" feature. Is it more suitable here, or I should leave it as is?
```

Comment:
Hmmm. Good question. I will have to think about this -- VALUES seems slightly better. But I will need to think through consequences and see what existing documentation says.